games/rpg/src/lib.rs

```diff
@@ -0,0 +1,1483 @@
+                else if class == 3 { dec!("4") }
+                else if class == 4 { dec!("0") }
+                else { dec!(".2")
+            };
+            let m1 = if class == 1 { dec!(0) }
+                else if class == 2 { dec!(15) }
+                else if class == 3 { dec!(15) }
+                else if class == 4 { dec!(20) }
+                else { dec!(20)
+            };
+            let m2 = if class == 1 { dec!("0") }
+                else if class == 2 { dec!(".5") }
+                else if class == 3 { dec!(".4") }
+                else if class == 4 { dec!("1.2") }
+                else { dec!(".2")
+            };
+            let phys_base = p1 * self.seed_50();
+            let phys_scale = p2 * self.seed_50();
+            let mag_base = m1 * self.seed_50();
+            let mag_scale = m2 * self.seed_50();
+            let ability_scale = dec!(0) * self.seed_50();
+            let weapon_data = Weapon {  
+                class: 1,
+                name: class_chosen,
+                level: dec!(1), 
+                physical_base: phys_base,
+                physical_scaling: phys_scale,
+                spell_base: mag_base,
+                spell_scaling: mag_scale,
+                ability: dec!(0),
+                ability_scaling: ability_scale,
+                range: dec!(1),
+                version: dec!(1),
+            };
+            let new_weapon = self.system_vault.authorize(||
+                borrow_resource_manager!(self.item_nft)
+                    .mint_non_fungible(&NonFungibleId::random(), weapon_data));
+            
+            self.system_vault.authorize(|| 
+                gold.burn());
+            self.system_vault.authorize(||
+                resource1.burn());
+            self.system_vault.authorize(||
+                resource2.burn());
+            self.system_vault.put(key_bucket);
+            new_weapon
+        }
+        pub fn create_armor_1(&mut self, gold: Bucket, resource1: Bucket, resource2: Bucket, part_chosen: u8) -> Bucket {
+            assert!(gold.resource_address() == self.token_gold);
+            assert!(resource1.resource_address() == self.token_greavite);
+            assert!(resource2.resource_address() == self.token_wood);
+            let key_bucket: Bucket = self.system_vault.take(1);
+            let part = if part_chosen == 1 {
+                String::from("Helmet")
+            }
+            else if part_chosen == 2 {
+                String::from("Chest")
+            }
+            else {
+                String::from("Pants")
+            };
+            let hp = dec!("5") * self.seed_50();
+            let def = dec!(".05") * self.seed_50();
+            let hp_bonus = dec!("10") * self.seed_50();
+            let def_bonus = dec!(".1") * self.seed_50();
+            let load = dec!(".01") * self.seed_50();
+            let ability_scale = dec!(0) * self.seed_50();
+            if self.seed_10() < dec!(5) {
```

Comment:
Branching off here leads you to repeating the code twice. However, there are ways to simplify this to make sure that code is not being repeated.